packages/services/src/terminal/default.ts

```diff
@@ -240,6 +275,9 @@ class DefaultTerminalSession implements TerminalSession.ISession {
   private _url: string;
   private _ws: WebSocket | null = null;
   private _noOp = () => { /* no-op */};
+
+  private _autoReconnectAttempt = 0;
+  private _autoReconnectLimit = 7;
```

Comment:
Ah I see, this is the upper bound 😄 

Perhaps instead of limiting the number of attempts, it should just keep trying as long as it is not disposed.

Reply:
my intention is to keep the same trial strategy as how the `kernel` deal with websocket reconnection, so that it might be easier to maintain :)